roles/rsyslog/tasks/main_core.yml

```diff
@@ -10,25 +10,38 @@
     - not __logging_is_ostree | d(false)
   block:
     # it is assumed that the only packages providing config files that might
-    # be modified are the base packages - if this is not so, then additional
-    # packages will need to be added to this check
+    # be modified are the rsyslog related base packages (but not others like iproute);
+    # if this is not so, then additional packages will need to be added to this check
     - name: Get status of rsyslog packages
       command: rpm -V {{ item }}  # noqa command-instead-of-module
-      loop: "{{ __rsyslog_base_packages }}"
+      loop: "{{ __rsyslog_base_packages | select('search', 'rsyslog') | list }}"
       register: __rsyslog_package_status
       failed_when: false
       changed_when: false
 
-    - name: Reset original confs - logging package is absent
-      package:
-        name: "{{ __rsyslog_base_packages }}"
+    - name: Clean up old rsyslog main configuration
+      file:
+        path: /etc/rsyslog.conf
+        state: absent
+
+    - name: Scan for old rsyslog configuration drop-ins
```

Comment:
This will remove drop-ins provided by other packages.  For example, cloud-init provides a drop-in for /etc/rsyslog.d.  I think there is some logic somewhere else in the role for scanning for drop-in files.

Reply:
In principle I don't understand how that can ever work -- you *cannot* just restore rsyslog.conf without also cleaning up the drop-ins, as you will get colliding options. A previous run of the role will replace the "big" rsyslog.conf with a small stub and move standard options to rsyslog.d/00-*.conf, which causes all these failures about duplicate options.

In other words, I'm stumped how this works in main -- I have the suspicion that *something* cleans up the drop-ins. (aside from removing the package also being broken)

Reply:
ah, understand -- I'll remove them more specifically then. (I initially just tried to remove 00-default.conf, but that's not sufficient)

Reply:
Reworked to use the existing config purging code that already existed further down -- I moved it upwards, and simplified it (as at that point the `__rsyslog_current_files` concept isn't relevant yet, and in fact would actually break the cleaning).